csharp/src/Drivers/Apache/Hive2/HiveServer2Statement.cs

```diff
@@ -255,5 +320,120 @@ protected void ValidateOptions(IReadOnlyDictionary<string, string> properties)
                 }
             }
         }
+
+        private async Task<QueryResult> ExecuteMetadataCommandQuery(CancellationToken cancellationToken)
+        {
+            return SqlQuery?.ToLowerInvariant() switch
+            {
+                GetCatalogsCommandName => await GetCatalogsAsync(cancellationToken),
+                GetSchemasCommandName => await GetSchemasAsync(cancellationToken),
+                GetTablesCommandName => await GetTablesAsync(cancellationToken),
+                GetColumnsCommandName => await GetColumnsAsync(cancellationToken),
+                GetPrimaryKeysCommandName => await GetPrimaryKeysAsync(cancellationToken),
+                GetCrossReferenceCommandName => await GetCrossReferenceAsync(cancellationToken),
+                null => throw new ArgumentNullException(nameof(SqlQuery), $"Metadata command for property 'SqlQuery' must not be empty or null. Supported metadata commands: {SupportedMetadataCommands}"),
+                "" => throw new ArgumentNullException(nameof(SqlQuery), $"Metadata command for property 'SqlQuery' must not be empty or null. Supported metadata commands: {SupportedMetadataCommands}"),
```

Comment:
I believe it's possible to say `null or ""` as the condition in a switch expression.